### PR TITLE
Documentation improvements - RTL and others

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: lib/ui/text.dart [hunk 1]
```diff
@@ -559,6 +559,82 @@ class ParagraphStyle {
 }
 
 /// A direction in which text flows.
+///
+/// ## Design discussion
+///
+/// Flutter is designed to address the needs of applications written in any of
+/// the world's languages. Some languages are written in from the left to the
```
Comment (cbracken) [inline lib/ui/text.dart]: written from

File: lib/ui/text.dart [hunk 2]
```diff
@@ -559,6 +559,82 @@ class ParagraphStyle {
...
+/// right (for example, English, Tamil, or Chinese), while others are written
+/// from the right to the left (for example Aramaic, Hebrew, or Urdu). Some are
+/// also written in a mixture, for example Arabic is mostly written
+/// right-to-left, with numerals written left-to-right.
+///
+/// (Flutter does not support other writing modes, such as vertical text or
+/// boustrophedon text, as these are rarely used in computer programs.)
```
Comment (cbracken) [inline lib/ui/text.dart]: Consider dropping the brackets as this stands on its own just fine.

File: lib/ui/text.dart [hunk 3]
```diff
@@ -559,6 +559,82 @@ class ParagraphStyle {
...
+/// [TextAlign.start] and [EdgeInsetsDirectional.start].
+///
+/// In contrast, the visual variants are useful when the text direction is known
+/// and not affected by the reading direction. For example, an application
+/// giving driving directions might show a "turn left" arrow on the left and a
+/// "turn right" arrow on the right — and would do so whether the application
+/// was localized to French (left-to-right) or Hebrew (right-to-left).
+///
+/// In practice, it is also expected that many developers will only be
+/// targetting one language, and in that case it may be simpler to think in
```
Comment (cbracken) [inline lib/ui/text.dart]: targeting

**Comment by cbracken** [comment]
lgtm

lgtm modulo nitpicks.

File: lib/ui/text.dart [hunk 4]
```diff
@@ -559,6 +559,82 @@ class ParagraphStyle {
...
+/// that only manifest in right-to-left environments.
+///
+/// In an effort to minimize the extent to which Flutter experiences this same
+/// category of issues, the lowest levels of the Flutter framework do not have a
+/// default text reading direction. Any time a reading direction is necessary,
+/// for example when text is to be displayed, or when a
+/// writing-direction-dependent value is to be interpreted, the reading
+/// direction must be explicitly specified. Where possible, such as in `switch`
+/// statements, the right-to-left case is listed first, to avoid the impression
+/// that it is an afterthought.
```
Comment (abarth) [inline lib/ui/text.dart]: It's also the first value in the enum.

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
I tweaked the wording a bit.

I'd like to add a bit more tutorial-flavoured text, but I don't really know what would be useful to someone who reaches this section.

Will land on green unless I hear otherwise.

**Pull request merged**